### PR TITLE
Added Advanced Search use-case.

### PR DESCRIPTION
**Pull request by miltonlog**

### CODE REVIEW

File: docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb [hunk 1]
```diff
@@ -350,18 +351,131 @@
...
+    "voyages_discharge_events = pd.DataFrame(v.__dict__ for v in voyages_discharge_events)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_voyage_load_area(voyage_events):\n",
```
Comment (abelesiotis-signal) [inline docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb:423]: You can define all the helper functions once in a cell before the initial get voyages section and then employ in both use cases, and avoid multiple declarations of the same functions.

File: docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb [hunk 2]
```diff
@@ -350,18 +351,131 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "voyages_load_events = api.get_voyages_by_advanced_search(vessel_class_id=vlcc_id, first_load_arrival_date_from=date_from,\n",
+    "                                                    event_horizon='Historical', event_purpose='Load')\n",
```
Comment (abelesiotis-signal) [inline docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb:401]: Minor alignment issue on line break.

File: docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb [hunk 3]
```diff
@@ -350,18 +351,131 @@
...
+    "\n",
+    "voyages_discharge_events = api.get_voyages_by_advanced_search(vessel_class_id=vlcc_id, first_load_arrival_date_from=date_from,\n",
+    "                                                    event_horizon='Historical', event_purpose='Discharge')"
```
Comment (abelesiotis-signal) [inline docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb:404]: Minor alignment issue on line break.

File: docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb [hunk 4]
```diff
@@ -350,18 +351,131 @@
...
+      "text/plain": [
+       "<Figure size 864x360 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plt.rcParams[\"figure.figsize\"] = (12,5)\n",
```
Comment (abelesiotis-signal) [inline docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb:472]: You can use `fig, ax = plt.subplots(figsize=(12, 5))` to modify the size of this figure only.

File: docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb [hunk 5]
```diff
@@ -350,18 +351,131 @@
...
+   "metadata": {},
+   "source": [
+    "### Advanced Voyage Search: Discharge origins"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This use-case demonstrates how to utilise the advanced search endpoint to extract historical voyages by **vessel_class_id**, **first_load_arrival_date** and a specific **event_purpose**. The voyages with the provided purpose can then be merged and filtered with a specific *load_area* and *discharge_country* in order to visualize the vessel flows into the specified country."
```
Comment (abelesiotis-signal) [inline docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb:379]: To increase readability, you can use the same markdown annotation for all parameters. Perhaps with inline code span (i.e. Back quotation mark " ` ") as these refer to variables.

Comment (abelesiotis-signal) [inline docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb:379]: The use of `event_purpose` is a bit confusing here. When this parameter is provided, only voyages that include at least one event of this type will be returned. The set of voyages with at least one discharge event will contain all voyages with at least one load event. Therefore, the call to get the voyages with a load event may be omitted. Also, the wording may change here to reflect that by setting this parameter to `'Discharge'`, we request all voyages with at least one discharge event.

**Comment by miltonlog** [comment]
@abelesiotis-signal Hello Ari and thank you for your extensive feedback. I have made the changes that you requested and removed  the `event_purpose` discrepancy which you correctly pointed out. Please let me know of any further changes.

File: docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb [hunk 6]
```diff
@@ -83,63 +83,62 @@
...
   {
    "cell_type": "code",
-   "execution_count": 9,
+   "execution_count": 8,
    "metadata": {},
    "outputs": [],
    "source": [
-    "def get_voyage_load_country(voyage_events):\n",
-    "    return next((e.country for e in voyage_events or [] if e.purpose=='Load'), None)\n",
     "\n",
```
Comment (abelesiotis-signal) [inline docs/examples/jupyter/VoyagesAPI/VoyagesAPI-UseCases.ipynb:139]: Please remove empty first line in the cell here.

**Review by abelesiotis-signal — APPROVED** [review]

**Review by abelesiotis-signal — APPROVED** [review]

**Pull request merged**